apps/mark-scan/backend/src/custom-paper-handler/state_machine.ts

```diff
@@ -705,6 +705,11 @@ export function buildMachine(
               entry: ['ejectPaperToFront'],
               on: {
                 NO_PAPER_ANYWHERE: 'resetting_state_machine_after_success',
+                // Sometimes paper ejected to front is not held by the motors but
+                // will still trigger input sensors.
+                // In this case we still want to allow the machine to progress.
+                PAPER_READY_TO_LOAD: 'resetting_state_machine_after_success',
+                PAPER_IN_INPUT: 'resetting_state_machine_after_success',
```

Comment:
These events will file before the paper is fully ejected. In theory, I might want to check that paper is detected in the input but also that paper is no longer detected inside the scanners (CIS sensors, etc.) to exit the `ejectPaperToFront` state, otherwise we will prematurely exit the state. Practically, may not matter.

It definitely doesn't matter for solving the problem you're trying to solve, so not blocking on it.

Reply:
Great catch. Recapping offline conversation - we should fix this in the long run but it introduces complication to the state machine because
* the way we convert printer/scanner status to events makes it difficult to rely solely on event type for union and intersection statuses
* checking in the state machine requires adding a new state because guarded transitions don't support returning a promise, and getting device status is async